src/Letter.ts

```diff
@@ -0,0 +1,119 @@
+
+    const content_array: Content[] = [];
+    text_array.forEach(function (slice, i) {
+        switch (slice) {
+            case 'bold>':
+                content_array.push({ text: text_array[i + 1], bold: true });
+                delete text_array[i + 1];
+                break;
+            case 'italic>':
+                content_array.push({ text: text_array[i + 1], italics: true });
+                delete text_array[i + 1];
+                break;
+            case '':
+                break;
+            default:
+                content_array.push(slice);
+        }
+    });
+
+    return content_array;
+};
+
+export class Letter {
+    doc: TDocumentDefinitions;
+    layout: LayoutFunction;
+    props: LetterProps;
+
+    constructor(props: LetterProps, layout_function?: LayoutFunction) {
+        this.doc = { content: '' };
+
+        this.layout = layout_function || dinLayoutFunction;
+        this.props = {
+            sender_address: [],
+            recipient_address: [],
+            information_block: '',
+            subject: '',
+            content: '',
+            signature: { type: 'text', name: '' },
+        };
+        this.setProps(props);
+    }
+
+    /** Render the letter as a string, used e.g. for emails. */
+    toString() {
+        const information_block = Array.isArray(this.props.information_block)
+            ? this.props.information_block.join('\n')
+            : this.props.information_block;
+        return information_block + '\n\n' + stripTags(this.props.content) + '\n' + this.props.signature['name'];
+    }
+
+    setProps(props: LetterProps) {
+        for (const i in props) {
+            // This is obviously fine but TypeScript isn't smart enough to figure that and I'm too lazy to teach it. :D
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            if (Object.prototype.hasOwnProperty.call(this.props, i)) this.props[i] = props[i];
+        }
+
+        this.doc = this.layout(
+            formatAddress(this.props.sender_address, ' • '),
+            formatAddress(this.props.recipient_address),
+            this.props.information_block,
+            this.props.subject,
+            Letter.parseTags(this.props.content),
+            Letter.handleSignature(this.props.signature)
+        );
+    }
+
+    /** Convert a given signature into a format understood by pdfmake. */
+    static handleSignature(signature: Signature): PdfMakeSignature | PdfMakeSignature[] | undefined {
+        if (!signature) return undefined;
+        switch (signature.type) {
+            case 'text':
+                return { text: signature.name, marginTop: mm2pt(2) };
+            case 'image':
+                return [
+                    { image: signature.value, marginTop: mm2pt(5) },
+                    { text: signature.name, marginTop: mm2pt(1) },
+                ];
+            default:
+                return undefined;
```

Comment:
This should really be `never`, if I understand TypeScript correctly.

Reply:
Oh yeah, the default case now isn't necessary anymore (and JS would've returned `undefined` anyway…).